### PR TITLE
Draft 27

### PR DESCRIPTION
**Pull request by ghedo**
Will need to wait until -27 is submitted, but for now use the -26 version number.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3762,10 +3818,124 @@ mod tests {
...
+                .unwrap();
+        assert_eq!(raw_params.len(), 73);
+
+        let new_tp =
+            TransportParams::decode(&mut raw_params, PROTOCOL_VERSION, false)
+                .unwrap();
+
+        assert_eq!(new_tp, tp);
+
+        // Clien encodes, server decodes.
```
Comment (LPardue) [inline src/lib.rs]:
```suggestion
        // Client encodes, server decodes.
```

Comment (ghedo) [reply src/lib.rs]: Fixed, thanks.

File: src/lib.rs [hunk 2]
```diff
@@ -3389,7 +3480,13 @@ impl TransportParams {
         let out_len = {
             let mut b = octets::Octets::with_slice(out);
 
-            b.put_u16(params_len as u16)?;
+            // TODO: once we drop support for drafts < 26, we should simplify
+            // the encoding to serialize the parameters into the output buffer
+            // directly.
+            if version < PROTOCOL_VERSION_DRAFT27 {
+                b.put_u16(params_len as u16)?;
```
Comment (LPardue) [inline src/lib.rs]: Nit: mention of draft 26 in comment

Comment (ghedo) [reply src/lib.rs]: Fixed.

File: extras/nginx/README.md [hunk 3]
```diff
@@ -89,7 +89,7 @@ http {
         ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
 
         # Add Alt-Svc header to negotiate HTTP/3.
-        add_header alt-svc 'h3-25=":443"; ma=86400';
+        add_header alt-svc 'h3-27=":443"; ma=86400';
```
Comment (junhochoi) [inline extras/nginx/README.md]: Also, there is `NGX_HTTP_V3_ALPN_ADVERTISE` defined in nginx patch but seems not used anywhere.

```
nginx-1.16.patch:+#define NGX_HTTP_V3_ALPN_ADVERTISE       "\x05h3-18"
```

**Comment by junhochoi** [comment]
LGTM

**Pull request merged**